### PR TITLE
add customizations for apps.krusie.io/v1alpha1/AdvancedCronJob

### PR DESCRIPTION
**Pull request by yike21**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Add third-party resources `apps.krusie.io/v1alpha1/AdvancedCronJob` into Resource Interpreter framework.

**Which issue(s) this PR fixes**:
Part of #3331

**Special notes for your reviewer**:
@Poor12 

**Does this PR introduce a user-facing change?**:
NONE

```release-note
`Resource Interpreter`: Support `apps.krusie.io/v1alpha1/AdvancedCronJob`
```

### CODE REVIEW

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/CronJob/customizations.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            if statusItems[i].status ~= nil and statusItems[i].status.lastScheduleTime ~= nil then
+              lastScheduleTime = statusItems[i].status.lastScheduleTime
+            end
+          end
+          desiredObj.status.active = active
+          desiredObj.status.type = type
+          desiredObj.status.lastScheduleTime = lastScheduleTime
+          return desiredObj
+        end
+    healthInterpretation:
```
Comment (yike21) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/CronJob/customizations.yaml:41]: For AdvancedCronJob, its `.spec.template` is either a `jobTemplate` or a `broadcastJobTemplate`, and `.status.type` indicates the type of the template, which is the only one considered legal in the health judgment.

**Comment by yike21** [comment]
The test report based on `karmadactl interpret` is below:
```shell
### Check the customizations in file
byk@debian:~/go/src/karmada/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/CronJob$ karmadactl interpret -f customizations.yaml --check
I0414 08:51:55.438502 2922048 deploy.go:75] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 08:51:55.438713 2922048 enable_option.go:63] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 08:51:55.451554 2922048 cmdinit.go:149] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 08:51:55.452072 2922048 register.go:160] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
-----------------------------------
SOURCE: declarative-configuration-advancedcronjob
TARGET: apps.kruise.io/v1alpha1 AdvancedCronJob   
RULERS:
    Retain:                                       UNSET
    InterpretReplica:                             UNSET
    ReviseReplica:                                UNSET
    InterpretStatus:                              UNSET
    AggregateStatus:                              PASS
    InterpretHealth:                              PASS
    InterpretDependency:                          PASS

### Execute the healthInterpretation rule
byk@debian:~/go/src/karmada/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/CronJob$ karmadactl interpret -f customizations.yaml --operation interpretHealth --observed-file testdata/observed-acj-nginx.yaml 
I0414 08:52:24.606929 2922115 deploy.go:75] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 08:52:24.607072 2922115 enable_option.go:63] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 08:52:24.616850 2922115 cmdinit.go:149] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 08:52:24.616988 2922115 register.go:160] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
---
# [1/1] healthy:
true

### Execute the dependencyInterpretation rule
byk@debian:~/go/src/karmada/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/CronJob$ karmadactl interpret -f customizations.yaml --operation interpretDependency --observed-file testdata/observed-acj-nginx.yaml 
I0414 08:52:52.673098 2922156 deploy.go:75] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 08:52:52.673233 2922156 enable_option.go:63] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 08:52:52.682118 2922156 cmdinit.go:149] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 08:52:52.682389 2922156 register.go:160] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
---
# [1/1] dependencies:
- apiVersion: v1
  kind: ConfigMap
  name: my-sample-config
  namespace: test-acj
- apiVersion: v1
  kind: ConfigMap
  name: mysql-config
  namespace: test-acj

### Execute the statusAggregation rule
byk@debian:~/go/src/karmada/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/CronJob$ karmadactl interpret -f customizations.yaml --operation aggregateStatus --observed-file testdata/observed-acj-nginx.yaml --status-file testdata/status-file.yaml 
I0414 08:53:18.181032 2922226 deploy.go:75] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 08:53:18.181331 2922226 enable_option.go:63] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 08:53:18.203768 2922226 cmdinit.go:149] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 08:53:18.204235 2922226 register.go:160] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
---
# [1/1] aggregatedStatus:
apiVersion: apps.kruise.io/v1alpha1
kind: AdvancedCronJob
metadata:
    labels:
        app: sample
    name: sample
    namespace: test-acj
spec:
    schedule: '*/2 * * * *'
    template:
        broadcastJobTemplate:
            spec:
                completionPolicy:
                    type: Never
                template:
                    metadata:
                        labels:
                            app: sample
                    spec:
                        containers:
                            - env:
                                - name: logData
                                  valueFrom:
                                    configMapKeyRef:
                                        key: log
                                        name: mysql-config
                                - name: lowerData
                                  valueFrom:
                                    configMapKeyRef:
                                        key: lower
                                        name: mysql-config
                              image: nginx:alpine
                              name: nginx
                        restartPolicy: Never
                        volumes:
                            - configMap:
                                name: my-sample-config
                              name: configmap
status:
    active:
        - apiVersion: apps.kruise.io/v1alpha1
          kind: BroadcastJob
          name: sample-1681378080
          namespace: test-acj
          resourceVersion: "3636404"
          uid: f96013ef-2869-49d7-adf3-8f7231cc5e2a
        - apiVersion: apps.kruise.io/v1alpha1
          kind: BroadcastJob
          name: sample-1681378080
          namespace: test-acj
          resourceVersion: "3635081"
          uid: d1f3c194-d650-4cce-b23d-307a445bb92e
    lastScheduleTime: "2023-04-13T09:30:00Z"
    type: BroadcastJob

```

**Comment by yike21** [comment]
The test report based on `karmadactl interpret` is below:
```shell
### Check the customizations in file
byk@debian:~/go/src/karmada/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/CronJob$ karmadactl interpret -f customizations.yaml --check
I0414 15:24:18.047536 2954857 deploy.go:75] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 15:24:18.047705 2954857 enable_option.go:63] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 15:24:18.056998 2954857 cmdinit.go:149] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 15:24:18.057131 2954857 register.go:160] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
-----------------------------------
SOURCE: declarative-configuration-advancedcronjob
TARGET: apps.kruise.io/v1alpha1 AdvancedCronJob   
RULERS:
    Retain:                                       UNSET
    InterpretReplica:                             UNSET
    ReviseReplica:                                UNSET
    InterpretStatus:                              UNSET
    AggregateStatus:                              PASS
    InterpretHealth:                              UNSET
    InterpretDependency:                          PASS

### Execute the statusAggregation rule
byk@debian:~/go/src/karmada/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/CronJob$ karmadactl interpret -f customizations.yaml --operation aggregateStatus --observed-file testdata/observed-acj-nginx.yaml --status-file testdata/status-file.yaml 
I0414 15:24:44.501376 2954911 deploy.go:75] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 15:24:44.501563 2954911 enable_option.go:63] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 15:24:44.513731 2954911 cmdinit.go:149] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 15:24:44.514235 2954911 register.go:160] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
---
# [1/1] aggregatedStatus:
apiVersion: apps.kruise.io/v1alpha1
kind: AdvancedCronJob
metadata:
    labels:
        app: sample
    name: sample
    namespace: test-acj
spec:
    schedule: '*/2 * * * *'
    template:
        broadcastJobTemplate:
            spec:
                completionPolicy:
                    type: Never
                template:
                    metadata:
                        labels:
                            app: sample
                    spec:
                        containers:
                            - env:
                                - name: logData
                                  valueFrom:
                                    configMapKeyRef:
                                        key: log
                                        name: mysql-config
                                - name: lowerData
                                  valueFrom:
                                    configMapKeyRef:
                                        key: lower
                                        name: mysql-config
                              image: nginx:alpine
                              name: nginx
                        restartPolicy: Never
                        volumes:
                            - configMap:
                                name: my-sample-config
                              name: configmap
status:
    active:
        - apiVersion: apps.kruise.io/v1alpha1
          kind: BroadcastJob
          name: sample-1681378080
          namespace: test-acj
          resourceVersion: "3636404"
          uid: f96013ef-2869-49d7-adf3-8f7231cc5e2a
        - apiVersion: apps.kruise.io/v1alpha1
          kind: BroadcastJob
          name: sample-1681378080
          namespace: test-acj
          resourceVersion: "3635081"
          uid: d1f3c194-d650-4cce-b23d-307a445bb92e
    lastScheduleTime: "2023-04-13T09:30:00Z"
    type: BroadcastJob

### Execute the dependencyInterpretation rule
byk@debian:~/go/src/karmada/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/CronJob$ karmadactl interpret -f customizations.yaml --operation interpretDependency --observed-file testdata/observed-acj-nginx.yaml 
I0414 15:25:07.628068 2954947 deploy.go:75] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 15:25:07.628536 2954947 enable_option.go:63] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 15:25:07.650315 2954947 cmdinit.go:149] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
I0414 15:25:07.651070 2954947 register.go:160] No default release version found. build version: version.Info{GitVersion:"", GitCommit:"c8720e81c7f4badb2a2e07ed7218391b3ea7d912", GitTreeState:"dirty", BuildDate:"2023-04-12T09:43:27Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
---
# [1/1] dependencies:
- apiVersion: v1
  kind: ConfigMap
  name: my-sample-config
  namespace: test-acj
- apiVersion: v1
  kind: ConfigMap
  name: mysql-config
  namespace: test-acj

```

**Comment by yike21** [comment]
make `git rebase` : force-pushed the `bundle-kruise-resources-cronjob` branch from 5845ddb to 9628830

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/CronJob/customizations.yaml (hunk 1 above)
Comment (Poor12) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/CronJob/customizations.yaml]: For `AdvancedCronJob`, status only reflects template information, not the status of actual workloads. So `interpretHealth` is not needed.

**Pull request merged**